### PR TITLE
set_par and process_qso updated

### PR DESCRIPTION
**Pull request by rezamonadi**
May you have a look plz...

### CODE REVIEW

File: c4_set_parameters.m [hunk 1]
```diff
@@ -77,11 +78,26 @@
...
+% model prior parameters
+
+prior_z_qso_increase = kms_to_z(30000);       % use QSOs with z < (z_QSO + x) for prior
+
+% instrumental broadening parameters
+width = 3;                                    % width of Gaussian broadening (# pixels)
+pixel_spacing = 1e-4;                         % wavelength spacing of pixels in dex
 
+% DLA model parameters: absorber range and model
+num_lines = 3;                                % number of members of the Lyman series to use
```
Comment (jibanCat) [inline c4_set_parameters.m]:
```matlab
num_lines = 2;                                % number of members of the CIV lines to use
```

File: c4_set_parameters.m [hunk 2]
```diff
@@ -77,11 +78,26 @@
...
 
-% I removed these functions since my code should work without them just in this stage
-% maybe add them later in the final version
+max_z_cut = kms_to_z(3000);                   % max z_DLA = z_QSO - max_z_cut
+max_z_c4 = @(wavelengths, z_qso) ...         % determines maximum z_DLA to search
+    (max(wavelengths) / civ_2_wavelength - 1) - max_z_cut;
```
Comment (jibanCat) [inline c4_set_parameters.m:94]:
```matlab
max_z_cut = kms_to_z(3000);                  % max z_CIV = z_QSO - max_z_cut
max_z_c4 = @(wavelengths, z_qso) ...         % determines maximum z_CIV to search
```
Does `civ_1_wavelength` represent CIV emission or `civ_2_wavelength`?

File: c4_set_parameters.m [hunk 3]
```diff
@@ -77,11 +78,26 @@
...
 
+min_z_cut = kms_to_z(3000);                   % min z_DLA = z_Ly∞ + min_z_cut
+min_z_c4 = @(wavelengths, z_qso) ...         % determines minimum z_DLA to search
+    max(min(wavelengths) / civ_2_wavelength - 1,                          ...
+        observed_wavelengths(civ_1_wavelength, z_qso) / civ_2_wavelength - 1 + ...
```
Comment (jibanCat) [inline c4_set_parameters.m]: I think the min_z_c4 should be lower?
The model range is 1410 ~ 1550, but you only search 1548~1550 for the absorbers.

suggested:
```matlab
        observed_wavelengths(1410, z_qso) / civ_2_wavelength - 1 + ...
```

File: c4_preload_qsos.m [hunk 4]
```diff
@@ -70,10 +70,10 @@
...
   this_noise_variance = this_noise_variance / this_median^2;
  
-  all_wavelengths{i}    =    this_wavelengths;%no longer (ind)
-  all_flux{i}           =           this_flux;
-  all_noise_variance{i} = this_noise_variance;
-  all_pixel_mask{i}     =     this_pixel_mask;
+  all_wavelengths{i,1}    =    this_wavelengths;%no longer (ind)
+  all_flux{i,1}           =           this_flux;
+  all_noise_variance{i,1} = this_noise_variance;
+  all_pixel_mask{i,1}     =     this_pixel_mask;
```
Comment (jibanCat) [inline c4_preload_qsos.m]: I think these modifications were the reasons why you need to do an extra transpose in the `process_qsos.m`

File: process_qsos.m [hunk 5]
```diff
@@ -115,54 +117,57 @@
...
+      log(                   this_num_c4) - log(this_num_quasars);
+  log_priors_no_c4(quasar_ind) = ...
+      log(this_num_quasars - this_num_c4) - log(this_num_quasars);
 
-  fprintf_debug('\n');
-  fprintf_debug(' ...     p(   DLA | z_QSO)        : %0.3f\n',     this_p_dla);
-  fprintf_debug(' ...     p(no DLA | z_QSO)        : %0.3f\n', 1 - this_p_dla);
+  fprintf('\n');
+  fprintf(' ...     p(   CIV | z_QSO)  mvn      : %0.3f\n',     this_p_c4);
+  fprintf(' ...     p(no CIV | z_QSO)        : %0.3f\n', 1 - this_p_c4);
```
Comment (jibanCat) [inline process_qsos.m:138]: debug printing can be turned on in the `set_parameters.m` without modifying

**Review by jibanCat — COMMENTED** [review]
I recently fixed a bug in the master branch of Roman's GP code: https://github.com/rmgarnett/gp_dla_detection/pull/4

Could you also fix this bug in your `process_qsos.m` code?

File: c4_set_parameters.m (hunk 2 above)
Comment (rezamonadi) [reply c4_set_parameters.m:94]: civ_1_wavelength and civ_2_wavelength are the doublet wavelengths. 
civ_1_wavelength = 1548.2049;
civ_2_wavelength =  1550.77845;

File: c4_set_parameters.m (hunk 3 above)
Comment (rezamonadi) [reply c4_set_parameters.m]: That's right

File: c4_set_parameters.m [hunk 6]
```diff
@@ -77,11 +78,26 @@
...
-% maybe add them later in the final version
+max_z_cut = kms_to_z(3000);                   % max z_DLA = z_QSO - max_z_cut
+max_z_c4 = @(wavelengths, z_qso) ...         % determines maximum z_DLA to search
+    (max(wavelengths) / civ_2_wavelength - 1) - max_z_cut;
 
+min_z_cut = kms_to_z(3000);                   % min z_DLA = z_Ly∞ + min_z_cut
+min_z_c4 = @(wavelengths, z_qso) ...         % determines minimum z_DLA to search
+    max(min(wavelengths) / civ_2_wavelength - 1,                          ...
+        observed_wavelengths(civ_1_wavelength, z_qso) / 1410 - 1 + ...
+        min_z_cut);
```
Comment (jibanCat) [inline c4_set_parameters.m:100]: If you do `observed_wavelengths(civ_1_wavelength, z_qso) / 1410 - 1` you will get `1548 * (1 + zQSO) / 1410 - 1`, which is even larger than zQSO while zCIV should be always lower than zQSO.

I think what you wanted was
```matlab
min_z_c4 = @(wavelengths, z_qso) ...         % determines minimum z_DLA to search
    max(min(wavelengths) / civ_2_wavelength - 1,                          ...
        observed_wavelengths(1410, z_qso) / civ_2_wavelength - 1 + ...
        min_z_cut);
```

**Pull request merged**